### PR TITLE
feat(es/lint): implement use-isnan rule

### PR DESCRIPTION
**Pull request by ArturAralin**
**Description:**

Implement use-isnan rule

**BREAKING CHANGE:**

No

### CODE REVIEW

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thanks!

File: crates/swc_ecma_lints/src/rules/use_is_nan.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                        }
+                    }
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Visit for UseIsNan {
```
[the remark this answers is not in the document]
Comment (ArturAralin) [reply crates/swc_ecma_lints/src/rules/use_is_nan.rs:117]: > I think this line should be removed?

Will it works without implementations Visit trait?

[the remark this answers is not in the document]
Comment (kdy1) [reply crates/swc_ecma_lints/src/rules/use_is_nan.rs:117]: Oh it's pointing different line.
I meant `noop_visit_type!()`

[the remark this answers is not in the document]
Comment (ArturAralin) [reply crates/swc_ecma_lints/src/rules/use_is_nan.rs:117]: Ok. I'll remove it

File: crates/swc/tests/errors/lints/use-is-nan/js/default/output.swc-stderr [hunk 2]
```diff
@@ -0,0 +1,72 @@
+
+  x Use the isNaN function to compare with NaN
+   ,----
+ 2 | if (foo == NaN) {}
```
Comment (ArturAralin) [inline crates/swc/tests/errors/lints/use-is-nan/js/default/output.swc-stderr:4]: @kdy1 seems like lines numeration is broken.

https://github.com/swc-project/swc/pull/4080/files#diff-1e3f349d2c56fc46384b18d56f835ba0ee7e04f1f3f32c489b58b6689ed9b5f7R1

Comment (kdy1) [reply crates/swc/tests/errors/lints/use-is-nan/js/default/output.swc-stderr:4]: Yeah, I'll fix it

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

---

swc-bump:
 - swc_ecma_lints

**Comment by kdy1** [comment]
Can you rebase?

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_lints

**Pull request merged**